templates/terraform/encoders/monitoring_slo.go.erb

```diff
@@ -0,0 +1,18 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2020 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+// Name/Service Level Objective ID is a query parameter and cannot
+// be given in data
+delete(obj, "sloId")
```

Comment:
I'm a little confused here. It seems like the user can specify the sloId but [it is being sent as `name`](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-3363-old..auto-pr-3363#diff-28c32f5906df20d338a2b0acd5bc2ee7R76-R81). Why does this need to be explicitly deleted from the obj?

Reply:
yeah, i could do this better but I wasn't sure the best way.  sloId (serviceLevelObjectiveId) is an optional query parameter on create that will get generated by the API, so we need to compute the value if it isn't given. url_param_only/input don't get read from the API response. I could remove this delete line since I don't think the API actually rejects it if given in data.